astropy/uncertainty/tests/test_distribution.py

```diff
@@ -374,3 +388,49 @@ def test_distr_noq_to_value():
     distr = ds.normal(10, n_samples=100, std=1)
     with pytest.raises(AttributeError):
         distr.to_value(u.m)
+
+
+def test_distr_angle():
+    # Check that Quantity subclasses decay to Quantity appropriately.
+    distr = Distribution([2., 3., 4.])
+    ad = Angle(distr, 'deg')
+    ad_plus_ad = ad + ad
+    assert isinstance(ad_plus_ad, Angle)
+    assert isinstance(ad_plus_ad, Distribution)
+
+    ad_times_ad = ad * ad
+    assert not isinstance(ad_times_ad, Angle)
+    assert isinstance(ad_times_ad, u.Quantity)
+    assert isinstance(ad_times_ad, Distribution)
+
+    ad += ad
+    assert isinstance(ad, Angle)
+    assert isinstance(ad, Distribution)
+    assert_array_equal(ad.distribution, ad_plus_ad.distribution)
+
+    with pytest.raises(u.UnitTypeError):
+        ad *= ad
+
+
+def test_distr_angle_view_as_quantity():
+    # Check that Quantity subclasses decay to Quantity appropriately.
+    distr = Distribution([2., 3., 4.])
+    ad = Angle(distr, 'deg')
+    qd = ad.view(u.Quantity)
+    assert not isinstance(qd, Angle)
+    assert isinstance(qd, u.Quantity)
+    assert isinstance(qd, Distribution)
+    # View directly as DistributionQuantity class.
+    qd2 = ad.view(qd.__class__)
+    assert not isinstance(qd2, Angle)
+    assert isinstance(qd2, u.Quantity)
+    assert isinstance(qd2, Distribution)
+    assert_array_equal(qd2.distribution, qd.distribution)
+    qd3 = ad.view(qd.dtype, qd.__class__)
+    assert not isinstance(qd3, Angle)
+    assert isinstance(qd3, u.Quantity)
+    assert isinstance(qd3, Distribution)
+    assert_array_equal(qd3.distribution, qd.distribution)
+
+    with pytest.raises(ValueError, match='just a new dtype'):
+        ad.view(np.dtype('f8'))
```

Comment:
This is confusing to me. Is it that this works for a regular Angle but *not* one that uses `Distribution`? Or it doesn't work for *any* angle? If it's the former I think it's probably a bug (which we can fix later to not hold up this PR, but should make an issue and add a comment here) whereas if it's the latter this should probably go in the regular `Angle` tests.

Reply:
Indeed, it works for `Angle` but cannot very easily work for any `Distribution` (since it has a structured `dtype`) - also just don't quite know what to do with things that change the number of elements (say, `.view('f4')`) - in practice, this basically never happens...

Anyway, I'll add a comment.

Reply:
Now added a comment and restructured the .view code as well.